client/src/components/grants/List.tsx

```diff
@@ -1,45 +1,44 @@
-import React, { useEffect } from 'react';
+import React, { useEffect } from "react";
 // import { RootState } from '../../reducers';
 import { Link } from "react-router-dom";
 import { RootState } from "../../reducers";
-import {
-  shallowEqual,
-  useSelector,
-  useDispatch,
-} from 'react-redux';
-import {
-  grantPath,
-} from "../../routes";
+import { shallowEqual, useSelector, useDispatch } from "react-redux";
+import { grantPath } from "../../routes";
 import { loadGrants, unloadGrants } from "../../actions/grants";
-import { GrantsListItem } from "../../reducers/grants";
 
 function GrantsList() {
   const dispatch = useDispatch();
-  const props = useSelector((state: RootState) => ({
-    loading: state.grants.loading,
-    grants: state.grants.grants
-  }), shallowEqual);
+  const props = useSelector(
+    (state: RootState) => ({
+      loading: state.grants.loading,
+      grants: state.grants.grants,
+      chainID: state.web3.chainID,
+    }),
+    shallowEqual
+  );
 
   useEffect(() => {
-    dispatch(loadGrants());
-    return () => {
-      dispatch(unloadGrants());
+    if (props.chainID) {
```

Comment:
why do we need to check chainID here? maybe we could change the `Layout` component to render the children only if wbe3 is initialized and chainID is set, so that we never have to check them in the rest of the app. wdyt @schultztimothy ?

Reply:
yeah `loadGrants()` will throw an error if chainId is undefined so I put that in to check before fetching. I like the idea of changing the Layout component though!